File: python/InnerClass.py
class Penutup:
    x = 10

    def printX(self):
        print(self.x)

    class Dalem:
        y = 20

        def printY(self):
            print(self.y)

obj1 = Penutup()
print(obj1.x)
obj1.printX()
# print(obj1.y) ==> Outer class tidak dapat mengakses propert/method inner class
# obj1.printY

obj2 = Penutup.Dalem()
# print(obj1.x) ==> inner class tidak dapat mengakses properti/method outer class
# obj1.printX()
print(obj2.y) 
obj2.printY()

File: cpp/InnerClass.cpp
#include <iostream>

class Penutup {
    private:
        int x = 10;

    

    public: 
        int getX() {
            return x;
        }

        class Dalem {
        private: 
            int y = 5;
        
        public:
            int getY() {
                return y;
            }
    };
};

int main() {
    Penutup tutup;

    std::cout << tutup.getX() << "\n"; //10
    // std::cout << tutup.getY() << "\n"; // Tidak memiliki akses ke properti maupun method inner class
    std::cout << typeid(tutup).name() << "\n"; // 7Penutup ==> artinya memiliki class penutup

    Penutup::Dalem isi; // 5
    // std::cout << isi.getX() << "\n"; // Tidak memiliki akses ke properti maupun method outer class
    std::cout << isi.getY() << "\n"; 
    std::cout << typeid(isi).name() << "\n";  // N7Penutup5DalemE ==> artinya memiliki class Dalem yg berada di class Penutup

}